### PR TITLE
extend target info

### PR DESCRIPTION
**Pull request by fishkerez**
## Description

Extend target info with the following properties:
Tags 
Image
instance type (micro, large, etc.)
Platform (linux distro)
add ../vpc/security-group to the location which is only the region currently
Launch time

## Type of Change

[* ] New Feature

### CODE REVIEW

File: ui_backend/pkg/rest/controllers.go [hunk 1]
```diff
@@ -79,15 +80,15 @@ func createRegionFindingsFromTargets(targets *backendmodels.Targets) []models.Re
...
+func getTargetRegion(target backendmodels.Target) (string, error) {
 	discriminator, err := target.TargetInfo.ValueByDiscriminator()
 	if err != nil {
 		return "", fmt.Errorf("failed to get value by discriminator: %w", err)
 	}
 
 	switch info := discriminator.(type) {
 	case backendmodels.VMInfo:
-		return info.Location, nil
+		return strings.Split(info.Location, "/")[0], nil
```
Comment (FrimIdan) [inline ui_backend/pkg/rest/controllers.go:91]: This is assuming you know how the location was created, and true only for AWS, is this safe/right to get the region?
Maybe better to also save the region in a separate value on the target if it is needed? Also, region is for AWS, what about other cloud-providers? I think we call it  `getTargetLocation` to be generic.

File: runtime_scan/pkg/provider/aws/instance.go [hunk 2]
```diff
@@ -32,21 +32,52 @@ type InstanceImpl struct {
...
 func (i *InstanceImpl) GetID() string {
 	return i.id
 }
 
 func (i *InstanceImpl) GetLocation() string {
-	return i.region
+	ret := i.region + "/" + i.vpcID
+	for _, group := range i.securityGroups { // TODO how do we want to represent multiple sg?
+		ret = ret + "/" + group
+	}
```
Comment (FrimIdan) [inline runtime_scan/pkg/provider/aws/instance.go:53]: maybe we should not put sg as part of the location, after all if instance can have multiple sg it doesn't say it exists in multiple locations.

File: runtime_scan/pkg/provider/aws/client.go [hunk 3]
```diff
@@ -398,15 +409,31 @@ func (c *Client) getInstancesFromDescribeInstancesOutput(result *ec2.DescribeIns
 				continue
 			}
 			ret = append(ret, &InstanceImpl{
-				ec2Client: c.ec2Client,
-				id:        *instance.InstanceId,
-				region:    regionID,
+				ec2Client:        c.ec2Client,
+				id:               *instance.InstanceId,
+				region:           regionID,
+				availabilityZone: *instance.Placement.AvailabilityZone,
```
Comment (FrimIdan) [inline runtime_scan/pkg/provider/aws/client.go:415]: are we sure `instance.Placement` or `instance.Placement.AvailabilityZone` are not nil?

Comment (akpsgit) [reply runtime_scan/pkg/provider/aws/client.go:415]: Actually, why do we even need AZ?

File: runtime_scan/pkg/provider/aws/instance.go (hunk 2 above)
Comment (akpsgit) [reply runtime_scan/pkg/provider/aws/instance.go:53]: it's not correct to concat the security groups as it's not one within the other. 
I think that it's better to move the Security groups out of the location into a new property of the instance (list of SGs). WDYT?

File: ui_backend/pkg/rest/controllers.go (hunk 1 above)
Comment (akpsgit) [reply ui_backend/pkg/rest/controllers.go:91]: This dashboard is specific to AWS. We can discuss if we want to use general terminology there.

Comment (FrimIdan) [reply ui_backend/pkg/rest/controllers.go:91]: if the widget is AWS specific maybe we should add it to the api path name or description

File: runtime_scan/pkg/provider/aws/instance.go (hunk 2 above)
Comment (FrimIdan) [reply runtime_scan/pkg/provider/aws/instance.go:53]: sgtm

File: runtime_scan/pkg/provider/aws/client.go (hunk 3 above)
Comment (akpsgit) [reply runtime_scan/pkg/provider/aws/client.go:415]: Agreed to add a validation function of the instance object, print the invalid field and skip the instance.

File: runtime_scan/pkg/types/types.go [hunk 4]
```diff
@@ -38,6 +39,11 @@ type ScanJobRunConfig struct {
 
 type Instance interface {
 	GetID() string
+	GetImage() string
+	GetType() string
+	GetLaunchTime() time.Time
+	GetPlatform() string
+	GetTags() []Tag
```
Comment (akpsgit) [inline runtime_scan/pkg/types/types.go:46]: Please add get SGs

File: ui_backend/pkg/rest/controllers.go (hunk 1 above)
Comment (akpsgit) [reply ui_backend/pkg/rest/controllers.go:91]: Need to move this to a function (with unit test). Need to do the split only for AWS provider, for other providers need to just return the location

File: api/openapi.yaml [hunk 5]
```diff
@@ -1453,6 +1453,11 @@ components:
           items:
             $ref: '#/components/schemas/Tag'
           nullable: true
+        securityGroups:
+          type: array
+          items:
+            $ref: '#/components/schemas/AwsSecurityGroup'
```
[the remark this answers is not in the document]
Comment (akpsgit) [reply api/openapi.yaml:1459]: Agree, we should create a new object.

File: runtime_scan/pkg/provider/aws/client.go [hunk 6]
```diff
@@ -419,17 +423,39 @@ func (c *Client) getInstancesFromDescribeInstancesOutput(result *ec2.DescribeIns
...
 	}
 	return ret
 }
 
-func getSecurityGroupsNames(sg []ec2types.GroupIdentifier) []string {
+func validateInstanceFields(instance ec2types.Instance) error {
+	if instance.InstanceId == nil {
+		return fmt.Errorf("instance id does not exist")
+	}
+	if instance.Placement == nil || instance.Placement.AvailabilityZone == nil {
```
Comment (akpsgit) [inline runtime_scan/pkg/provider/aws/client.go:441]: maybe we should split this if in 2 to distinguish?

**Review by akpsgit — APPROVED** [review]

**Pull request merged**